esmvalcore/_recipe.py

```diff
@@ -665,13 +696,21 @@ def get_matching(attributes):
 
 
 def _get_preprocessor_products(variables, profile, order, ancestor_products,
-                               config_user):
+                               config_user, fx_case=False):
     """Get preprocessor product definitions for a set of datasets."""
     products = set()
 
-    for variable in variables:
-        variable['filename'] = get_output_file(variable,
-                                               config_user['preproc_dir'])
+    if not fx_case:
+        for variable in variables:
+            variable['filename'] = get_output_file(variable,
+                                                   config_user['preproc_dir'])
+    else:
+        parent_variable = variables[1]
+        variables = [variables[0]]
+        for variable in variables:
+            variable['filename'] = get_output_file(parent_variable,
+                                                   config_user['preproc_dir'],
+                                                   fx_var_alias=variable)
```

Comment:
nope, that needs to stay a list of variables for some reason, otherwise the `filename` key does not get assigned

Reply:
To me, this looks equivalent. Could you please point out my mistake?

Reply:
because I am redefining the whole set of `variables` in this case, to be used below for updating all the settings and building the products:
```
    if not var_fxvar_coupling:
        for variable in variables:
            variable['filename'] = get_output_file(variable,
                                                   config_user['preproc_dir'])
    else:
        parent_variable = variables[1]
        variables = [variables[0]]
        for variable in variables:
            variable['filename'] = get_output_file(parent_variable,
                                                   config_user['preproc_dir'],
                                                   fx_var_alias=variable)

    if ancestor_products:
        grouped_ancestors = _match_products(ancestor_products, variables)
    else:
        grouped_ancestors = {}

    for variable in variables:
        settings = _get_default_settings(
            variable,
            config_user,
            derive='derive' in profile,
        )
...
```
variables are now `[variables[0]]`

Reply:
no mistake on yer part K-man, just a reassignment of variables->variables, you ok with it?

Reply:
Now I get it. I'm not super happy with the loop over clearly only one element (and am also wondering if there ever could be more than just two elements in the original variables), but :man_shrugging:.

Reply:
ya, looping on one element is peasant-y but it's the path of least changes I could find; the `variables` in the non-fx case could be a list of multiple elements, but not for fx since the task is built per fx variable. This will go to when we eliminate the ancestors, but until then it's the best I can muster for correct functionality :beer:

Reply:
I'm ok(ish) with the code as it stands. I made one last suggestion by updating the suggestion above. If you like, take it, otherwise leave as is. :beers:

Reply:
dude, you a hero - cheers for all the work, done and put this one last change :beer: